src/modules/producer.js

```diff
@@ -118,7 +119,7 @@ function checkRpc (queue, msg, options) {
 
         if (publishOrSendToQueue.call(this, queue, msg, options)) {
             //defered promise that will resolve when response is received
-            let responsePromise = Promise.defer();
+            let responsePromise =  new Deferred();
```

Comment:
Thing is, how do you replace it? Just using `new Promise` does not help because it should be resolve outside of the scope:

```js
function resolveIt(promise) {
  promise.resolve();
}

function getIt() {
  return new Promise();
}

const myPromise = getIt();
myPromise.then(function() {
  console.log('Someone resolved me! Hourra!');
});
```